src/components/TextField/TextField.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useState } from "react";
+
+interface ITextFieldProps {
+  customClassNames?: string;
+  value?: string;
+  label?: string;
+  type: string;
+  name?: string;
+  placeholder?: string;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
```

Comment:
Very typescripty, I am thinking instead of having to extract the text value from `event.target.value` higher up the component chain, why not pass it up from here?
```suggestion
  onChange: (value: string) => void;
  
  const handleTextInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
      props.onChange(event.target.value);
  }
```